File: 10_text_and_transformation/src/main/java/com/doyou/a10_text_and_transformation/MainActivity.java
package com.doyou.a10_text_and_transformation;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.os.Bundle;

import com.doyou.a10_text_and_transformation.view.CameraView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private CameraView mCameraView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mCameraView = findViewById(R.id.cameraView);

        ObjectAnimator bottomFlipAnimator = ObjectAnimator.ofFloat(mCameraView,"bottomFlip",30);
        bottomFlipAnimator.setDuration(1000);

        ObjectAnimator topFlipAnimator = ObjectAnimator.ofFloat(mCameraView,"topFlip",-30);
        topFlipAnimator.setStartDelay(200);
        topFlipAnimator.setDuration(1000);

        ObjectAnimator rotationFlipAnimator = ObjectAnimator.ofFloat(mCameraView,"flipRotation",270);
        rotationFlipAnimator.setStartDelay(200);
        rotationFlipAnimator.setDuration(1000);

        AnimatorSet animatorSet = new AnimatorSet();
        animatorSet.playSequentially(bottomFlipAnimator,rotationFlipAnimator,topFlipAnimator);
        animatorSet.setStartDelay(1000);
        animatorSet.start();
    }
}

File: app/src/main/java/com/hencoder/hencoder/MainActivity.kt
package com.hencoder.hencoder

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.hencoder.Utils


class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        window.decorView.postDelayed({
////            val animator1 = ObjectAnimator.ofInt(objectAnimatorView, "turnoverDegreeFirst", 0, 30)
////            animator1.duration = 1000
////            animator1.interpolator = LinearInterpolator()
////
////            val animator2 = ObjectAnimator.ofInt(objectAnimatorView, "degree", 0, 270)
////            animator2.duration = 2000
////            animator2.interpolator = AccelerateDecelerateInterpolator()
////
////            val animator3 = ObjectAnimator.ofInt(objectAnimatorView, "turnoverDegreeLast", 0, 30)
////            animator3.duration = 1000
////            animator3.interpolator = LinearInterpolator()
////
////            val animatorSet = AnimatorSet()
////            // 三个动画依次执行
//////                animatorSet.playSequentially(animator2, animator3);
////            animatorSet.playSequentially(animator1, animator2, animator3)
////            animatorSet.start()
//
//
//
//            var bottomFlipAnimator = ObjectAnimator.ofFloat(objectAnimatorView,"bottomFlip",30f)
//            bottomFlipAnimator.duration = 1000
//
//            var topFlipAnimator = ObjectAnimator.ofFloat(objectAnimatorView,"topFlip",-30f)
//            topFlipAnimator.startDelay = 200
//            topFlipAnimator.duration = 1000
//
//            var flipRotationAnimator = ObjectAnimator.ofFloat(objectAnimatorView,"flipRotation",270f)
//            flipRotationAnimator.startDelay = 200
//            flipRotationAnimator.duration = 1000
//
//            var animatorSet = AnimatorSet()
//            animatorSet.playSequentially(bottomFlipAnimator,flipRotationAnimator,topFlipAnimator)
//            animatorSet.startDelay = 1000
//            animatorSet.start()
//        }, 2000)

        Utils.shout()
    }
}
